src/ui-kit/BaseFiltration/useHotFilters.tsx

```diff
@@ -0,0 +1,63 @@
+import { useCallback, useMemo } from '@redneckz/uni-jsx/lib/hooks';
+import type { BlockDef } from '../../model/ContentPageDef';
+import { Badge } from '../Badge/Badge';
+import type { TabItemBase } from '../TabsControl/TabItem';
+import type { TabsControlProps } from '../TabsControl/TabsControlProps';
+import { getBadgeCount } from './getBadgeCount';
+import type { HotFilterButton } from './HotFilterButton';
+import { hotFilterRecordCmp } from './HotFilterRecord';
+
+const EMPTY_ARR = [];
+const EMPTY_OBJ = {};
+
+interface UseHotFiltersParams {
+  hotFilters?: HotFilterButton[];
+  filtrationState?: Record<string, string[]>;
+  blocks?: BlockDef[];
+  initialFiltrationState: Record<string, string[]>;
+  replaceFiltrationState: (_: Record<string, string[]>) => void;
+}
+
+export const useHotFilters = ({
+  hotFilters = EMPTY_ARR,
+  filtrationState = EMPTY_OBJ,
+  blocks = EMPTY_ARR,
+  initialFiltrationState = EMPTY_OBJ,
+  replaceFiltrationState,
+}: UseHotFiltersParams): TabsControlProps => {
+  const activeItemIndex = useMemo(
+    () =>
+      Math.max(
+        0,
+        hotFilters.findIndex((_) => hotFilterRecordCmp(_.filter, filtrationState)),
+      ),
+    [hotFilters, filtrationState],
+  );
+
+  const items: TabItemBase[] = useMemo(
+    () =>
+      hotFilters.map((_) => ({
+        text: _.text,
+        appendRight: (
+          <Badge
+            version={_ === hotFilters[activeItemIndex] ? 'primary' : 'secondary'}
+            count={getBadgeCount(blocks, _.filter)}
+          />
+        ),
+      })),
+    [hotFilters, blocks, activeItemIndex],
+  );
+
+  const handleChange = useCallback(
+    (_: TabItemBase) => {
+      replaceFiltrationState(hotFilters[items.indexOf(_)].filter || initialFiltrationState);
```

Comment:
может упростить вход таким образом:
```suggestion
      onChangeFiltrationState(hotFilters[items.indexOf(_)].filter);
```
тогда initialFiltrationState и replaceFiltrationState заменятся на onChangeFiltrationState